tests/unit_tests/run_models/test_base_run_model.py

```diff
@@ -120,3 +124,34 @@ def get_run_path_mock(realizations, iteration=None):
         run_path=run_path,
     )
     assert brm.check_if_runpath_exists() == expected
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+@pytest.mark.parametrize(
+    "run_path_format", ["realization-<IENS>/iter-<ITER>", "realization-<IENS>"]
+)
+def test_delete_run_path(run_path_format):
+    simulation_arguments = EnsembleExperimentRunArguments(
+        random_seed=None,
+        active_realizations=[True],
+        current_case=None,
+        target_case=None,
+        start_iteration=0,
+        iter_num=0,
+        minimum_required_realizations=0,
+        ensemble_size=1,
+        stop_long_running=False,
+    )
+    run_path = Path(run_path_format.replace("<IENS>", "0").replace("<ITER>", "0"))
+    os.makedirs(run_path)
+    assert run_path.exists()
+    model_config = ModelConfig(runpath_format_string=run_path_format)
+    subs_list = SubstitutionList()
+    config = MagicMock()
+    config.model_config = model_config
+    config.substitution_list = subs_list
+
+    brm = BaseRunModel(simulation_arguments, config, None, None, None, None)
+    brm.rm_run_path()
+    assert not run_path.exists()
+    assert run_path.parent.exists()
```

Comment:
Perhaps add an assert that any extra file/dir unknown to ERT alongside `realizations-*` is untouched? It is e.g. customary to have a `share` directory like this.